### ISSUE TITLE
Incorrect parsing of optionals and no warning that a critical parameter missing

### ISSUE THREAD

**Issue by schollii** [infracost/infracost#2406]
Save this in a file `main.tf` in a new folder: 
```hcl
variable "config_wrong_cost" {
  type = object({
    db_engine = optional(string, "postgres")
  })

  default = {}
}

variable "config_correct_cost" {
  type = object({
    db_engine = string
  })

  default = {db_engine = "postgres"}
}

resource "aws_db_instance" "postgres" {
  engine         = var.config_wrong_cost.db_engine // **
  # engine         = var.config_correct_cost.db_engine // **
  instance_class = "db.t3.2xlarge"
}
```
Cd to the folder. This passes `terraform validate` and requires terraform 1.4.4 and infracost v0.10.19. A `terraform plan` shows that the engine type is as expected ("postgres"). 

Now run `infracost breakdown --path .` on the above: will show 0 cost for the DB: 
```
$ infracost breakdown --path .
Evaluating Terraform directory at .
  ✔ Downloading Terraform modules 
  ✔ Evaluating Terraform directory 
  ✔ Retrieving cloud prices to calculate costs 

Project: .

 Name                                                        Monthly Qty  Unit   Monthly Cost 
                                                                                              
 aws_db_instance.postgres                                                                     
 ├─ Database instance (on-demand, Single-AZ, db.t3.2xlarge)          730  hours         $0.00 
 └─ Storage (general purpose SSD, gp2)                                20  GB            $2.30 
                                                                                              
 OVERALL TOTAL                                                                          $2.30 
──────────────────────────────────
1 cloud resource was detected:
∙ 1 was estimated, it includes usage-based costs, see https://infracost.io/usage-file
```

Finally switch which of the two `// **` lines is commented out, and re-run infracost. This time it shows correctly the cost of DB instance is non-zero: 
```
 ├─ Database instance (on-demand, Single-AZ, db.t3.2xlarge)          730  hours       $422.67 
```

Note how for the cost=0 case, there is no indication whatsoever that a parameter required (ie critical) to determine cost was not found by infracost. In my original code, the instance class is similarly incorrectly computed, and shows as `db_instance_class-mock` in the output. No indication of any problem there either, if you didn't notice the odd name you would have no way of knowing.

**Comment by hugorut** [comment]
Thanks @schollii, this will be resolved when https://github.com/infracost/infracost/pull/2389 is merged

**Comment by hugorut** [comment]
The fix for this is now live in the [v0.10.20](https://github.com/infracost/infracost/releases/tag/v0.10.20) release
